src/full/Agda/TypeChecking/Reduce.hs

```diff
@@ -920,6 +920,7 @@ appDefE'' v cls rewr es = traceSDoc "tc.reduce" 90 ("appDefE' v = " <+> pretty v
               nvars = size $ clauseTel cl
           -- if clause is underapplied, skip to next clause
           if length es < npats then goCls cls es else do
+            termCheck <- asksTC envTermCheckReducing
```

Comment:
Could this new line be moved under the `Yes` branch below?
To keep the diff small, you could use a multi-way if there...